cache/disk/disk.go

```diff
@@ -1040,7 +1044,7 @@ func (c *diskCache) Contains(ctx context.Context, kind cache.EntryKind, hash str
 
 	if c.proxy != nil {
 		exists, foundSize = c.proxy.Contains(ctx, kind, hash)
```

Comment:
If we know the size ahead of time, and it's larger than maxProxyBlobSize we could skip this call.